src/gen_pfc.c

```diff
@@ -469,18 +469,17 @@ static int _gen_pfc_arch(const struct db_filter_col *col,
  */
 int gen_pfc_generate(const struct db_filter_col *col, int fd)
 {
-	int rc = 0;
 	int newfd;
 	unsigned int iter;
 	FILE *fds;
 
 	newfd = dup(fd);
 	if (newfd < 0)
-		return errno;
+		return -ECANCELED;
```

Comment:
Random low-priority comment - when I was cscoping errno, I noticed some of our functions in util.c and scmp_bpf_disasm.c sometimes return `-errno` and sometimes `errno`.  The doxygen comments of the functions imply they all should be returning a negative error number.

Reply:
Hmm, good point.  We're not quite extending the same return value promise to these tools (yet), but we should fix that.  I'll add a patch to the next force push.

Reply:
Now that I'm looking at it, scmp_bpf_disasm.c looks correct, it's just the comments that are wrong.  For the API we want to return negative values on error, but for the terminal apps we want to return a positive errno value.

Does that make sense, or am I crazy?

Reply:
Yes, I agree with your analysis.  The use of positive errors on a terminal has always been a pet peeve of mine because it seems so... error prone :(.

Let's fix up the comments some time, but it's obviously not critical or related to this work.

Reply:
> ... a pet peeve of mine because it seems so... error prone :(

:)

> Let's fix up the comments some time, but it's obviously not critical or related to this work.

I thought I fixed the comments, but I'm sure I missed a few ... can you point me in the right direction?